### PR TITLE
Fix Japanese IME input

### PR DESCRIPTION
**Pull request by danielheffernan**
The previous fix for this (1040918) fixed some bugs for users on IMEs, but did not work when users typed quickly, and didn't work well with the expiry field.

r? @mduvall

### CODE REVIEW

**Comment by mduvall** [comment]
LGTM :+1:

File: src/jquery.payment.coffee [hunk 1]
```diff
@@ -150,12 +150,29 @@ safeVal = (value, $target) ->
     $target.prop('selectionStart', cursor)
     $target.prop('selectionEnd', cursor)
 
+# Replace Full-Width Chars
+
+replaceFullWidthChars = (str = '') ->
+  fullWidth = '\uff10\uff11\uff12\uff13\uff14\uff15\uff16\uff17\uff18\uff19'
```
Comment (mduvall) [inline src/jquery.payment.coffee]: Will assume this is 0-9 full width, thanks for adding!

Comment (danielheffernan) [inline src/jquery.payment.coffee]: Yup! The last half of the second byte corresponds to the numeral, if you want a quick way of identifying it. `'\uff14' == '４'`

**Comment by mduvall** [comment]
Question: Will we need to update any of the methods for `reformat.*`? 

Overall, if you tested by feel free to merge. I will merge this in the morning and deploy.

**Comment by jamesreggio** [comment]
I'd be happy to review this today if you'd like for me to do so.

**Comment by mduvall** [comment]
@jamesreggio !!! Yeah, that would be awesome

**Comment by jamesreggio** [comment]
Very interesting. I didn't realize that full-width chars were being input and then swallowed by the complicated logic. This works (and is an improvement) for WebKit and IE, and I suggest you merge it.

The field is still non-operational in Firefox without either `input="tel"` or `ime-mode: 'disable'` on the element. Unfortunately, Firefox neither allows you to cancel the IME from opening (which should be possible during the [`compositionstart` event](https://developer.mozilla.org/en-US/docs/Web/Events/compositionstart#Gecko_notes), but isn't), and it [will not restore the system-level IME to its original state](https://developer.mozilla.org/en-US/docs/Web/CSS/ime-mode#Notes) when `ime-mode: 'disable'` is used, which is highly annoying.

I modified the Stripe Checkout codebase to use `ime-mode: 'disable'` last Friday, which makes this workable (but inconvenient) in Firefox. Perhaps a better fix would be to use `type="tel"` for desktop browsers and sidestep the problem entirely.

**Comment by mduvall** [comment]
Great, sounds good, will merge and cut a patch release.

**Comment by danielheffernan** [comment]
Thank you, @jamesreggio! The Firefox situation isn't great but if the IME not restoring is the tradeoff, we're not in a bad position at all. The `type="tel"` on desktop is an interesting hack.

**Pull request merged**